src/components/richText/embeddedKeyNode/EmbeddedKeyNode.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  TextModeType,
+  EditorConfig,
+  TextNode, 
+  NodeKey,
+  SerializedLexicalNode,
+  Spread,
+} from "lexical";
+
+/**
+ * The raw Object obtained when first parsing a serialized {@link EmbeddedKeyNode}.
+ */
+export type SerializedEmbeddedKeyNode = Spread<
+  {
+    text: string,
+    rawText: string,
+    format: number,
+    style: string,
+    mode: TextModeType,
+    detail: number,
+    type: "embedkey",
+    version: 1,
+  },
+  SerializedLexicalNode
+>;
+
+/**
+ * Defines a Lexical Dev {@link TextNode} that supports embedded keys in Rich Text. 
+ * This Node is meant to be used with a read-only Lexical Editor. As such, it does 
+ * not have setters for its various attributes.
+ */
+export class EmbeddedKeyNode extends TextNode {
+  constructor(
+    text: string, 
+    rawText: string, 
+    key?: NodeKey
+  ) {
+    super(text, key);
+    this.__text = text;
+    this.__rawText = rawText;
+  }
+
+  static getType(): string {
+    return "embedkey";
+  }
+
+  static clone(node: EmbeddedKeyNode): EmbeddedKeyNode {
+    return new EmbeddedKeyNode(node.__text, node.__rawText, node.__key);
+  }
+
+  /**
+   * Inserts the {@link EmbeddedKeyNode}'s placeholder {@link HTMLElement} into the Lexical Dev's DOM.
+   * TODO: Completely emulate embedded key styling in entitiesstorm
+   */
+  createDOM(config: EditorConfig): HTMLElement {
+    const dom = super.createDOM(config);
+    dom.className = 
```

Comment:
should we move these to the default style tailwind file to configure the lexical theme?

Reply:
I tried using `embedkey` and `badge-info` and neither works. I believe it's because the name has to be in this list: https://github.com/facebook/lexical/blob/dcebd0a72d595da2d3e6fe30533b864f505f1bcc/packages/lexical/src/LexicalEditor.ts#L108